Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,525 @@
+            continue;
+        }
+
+        CFArrayAppendValue(fontFamilyNames.get(), name);
+    }
+
+    return static_cast<CFArrayRef>(CFAutorelease(fontFamilyNames.release()));
+}
+
+/**
+ * Helper method to retrieve names of individual fonts under a font family.
+ */
+CFArrayRef _DWriteGetFontNamesForFamilyName(CFStringRef familyName) {
+    woc::unique_cf<CFMutableArrayRef> fontNames(CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks));
+
+    ComPtr<IDWriteFactory> dwriteFactory;
+    RETURN_NULL_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
+
+    // Get the system font collection.
+    ComPtr<IDWriteFontCollection> fontCollection;
+    RETURN_NULL_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
+
+    // Get the font family.
+    CFIndex familyNameLength = CFStringGetLength(familyName);
+    std::vector<UniChar> unicharFamilyName(familyNameLength + 1);
+    CFStringGetCharacters(familyName, CFRangeMake(0, familyNameLength), unicharFamilyName.data());
+
+    size_t index = 0;
+    BOOL exists = false;
+
+    RETURN_NULL_IF_FAILED(fontCollection->FindFamilyName(reinterpret_cast<wchar_t*>(unicharFamilyName.data()), &index, &exists));
+    if (!exists) {
+        TraceError(TAG, L"Failed to find the font family name.");
+        return static_cast<CFArrayRef>(CFAutorelease(fontNames.release()));
+    }
+
+    ComPtr<IDWriteFontFamily> fontFamily;
+    RETURN_NULL_IF_FAILED(fontCollection->GetFontFamily(index, &fontFamily));
+
+    ComPtr<IDWriteFontList> fontList;
+    RETURN_NULL_IF_FAILED(
+        fontFamily->GetMatchingFonts(DWRITE_FONT_WEIGHT_THIN, DWRITE_FONT_STRETCH_UNDEFINED, DWRITE_FONT_STYLE_NORMAL, &fontList));
+
+    size_t count = fontList->GetFontCount();
+
+    for (size_t i = 0; i < count; i++) {
+        ComPtr<IDWriteFont> font;
+        RETURN_NULL_IF_FAILED(fontList->GetFont(i, &font));
+
+        ComPtr<IDWriteLocalizedStrings> fullName;
+        BOOL exist = FALSE;
+        RETURN_NULL_IF_FAILED(font->GetInformationalStrings(DWRITE_INFORMATIONAL_STRING_FULL_NAME, &fullName, &exist));
+
+        if (exist) {
+            CFStringRef name = _CFStringFromLocalizedString(fullName.Get());
+            if (CFStringGetLength(name) == 0) {
+                TraceError(TAG, L"Failed to convert the localized string to wide string.");
+                continue;
+            }
+
+            CFArrayAppendValue(fontNames.get(), name);
+        }
+    }
+
+    return static_cast<CFArrayRef>(CFAutorelease(fontNames.release()));
+}
+
+/**
+ * Helper method that maps a font name to the name of its family.
+ *
+ * Note: This function currently uses a cache, meaning that fonts installed during runtime will not be reflected
+ */
+CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
+    woc::unique_cf<CFLocaleRef> locale(CFLocaleCopyCurrent());
+
+    static CFDictionaryRef fontToFamilyMap = [&locale]() {
```

Comment:
>&locale [](start = 46, length = 7)

Seems weird to have a local variable be passed by ref to a one time init static lambda. My expectation would be that the locale would be static too if some static state needed it otherwise it would potentially get out of sync?